engine/pom.xml

```diff
@@ -218,8 +218,10 @@
                             org.hibernate.search.query;version="${project.version}",
                             org.hibernate.search.query.collector;version="${project.version}",
                             org.hibernate.search.query.dsl;version="${project.version}",
+                            org.hibernate.search.query.dsl.sort;version="${project.version}",
                             org.hibernate.search.query.engine;version="${project.version}",
                             org.hibernate.search.query.engine.spi;version="${project.version}",
+                            org.hibernate.search.query.engine.spi.impl;version="${project.version}",
```

Comment:
@gunnarmorling @gsmet I created a [JIRA ticket](https://hibernate.atlassian.net/browse/HSEARCH-2333) and [another PR](https://github.com/hibernate/hibernate-search/pull/1141) for this particular issue. Once it's merged, I'll rebase this PR on master, and there should no longer be a need for exporting an `impl` package.